azurelinuxagent/common/datacontract.py

```diff
@@ -0,0 +1,83 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+from azurelinuxagent.common.exception import ProtocolError
+import azurelinuxagent.common.logger as logger
+
+"""
+Data contract between guest and host
+"""
```

Comment:
There some imports referencing the old module, please do a global search of usages of these classes/methods.

E.g.
> from azurelinuxagent.common.protocol.restapi import DataContract, set_properties

in imds.py